lightning/restore/restore.go

```diff
@@ -339,19 +348,42 @@ func (rc *RestoreController) listenCheckpointUpdates(wg *sync.WaitGroup) {
 
 		lock.Unlock()
 
+		// Note: about gofailLabel1, gofailLabel2, etc.., just a hack way
+		// for avoiding generated error message e.g:
+		// `failpoint: "github.com/pingcap/tidb-lightning/lightning/restore/FailIfIndexEngineImported" got value 140 of type "int" but expected type "int"`
+
 		// gofail: var FailIfImportedChunk struct{}
 		// if _, ok := scp.merger.(*ChunkCheckpointMerger); ok {
 		// 	wg.Wait()
 		// 	panic("forcing failure due to FailIfImportedChunk")
 		// }
-		// continue
+		// goto gofailLabel1
+	gofailLabel1:
+		if false {
+			goto gofailLabel1
+		}
```

Comment:
You don't need these `if false` stuff, see the function `OpenEngine` in `kv/importer.go`.

Reply:
addressed